document projection clause changes (#1618)
neo4j/docs-cypher: modules/ROOT/pages/clauses/order-by.adoc

The variables available to `ORDER BY` depend on whether or not the preceding `RETURN` or `WITH` clause performs an aggregation to combine results or uses `DISTINCT` to remove duplicates.

> * If the `RETURN` or `WITH` is not aggregating values or using `DISTINCT`, then `ORDER BY` has access to the projected variables from the preceding `RETURN` or `WITH` clause.

Reviewer: No, this is still the same - if it is non-aggregating everything is visible

> * If the `RETURN` or `WITH` is not aggregating values or using `DISTINCT`, then `ORDER BY` has access to the projected variables from the preceding `RETURN` or `WITH` clause.

Author: ```suggestion
* If the `RETURN` or `WITH` is not aggregating values or using `DISTINCT`, then `ORDER BY` can reference any variables referenced in the preceding `RETURN` or `WITH` clause.
```
